content/articles/react-drf-shortener/index.md

```diff
@@ -0,0 +1,369 @@
+### How to Create a URL Shortener with React, Django Rest Framework, and SQLite Database
+
+This article covers creating a custom URL(Uniform Resource Locator) shortener service. The URL shortener will take in a long URL and provide an equivalent short URL suitable for posting on social media and sharing with friends. The shortener service will be made up of 3 parts; the front-end for interacting with the client(we will create the front-end using React), the backend for creating the server, and providing the API routes(we will create the backend using Django Rest Framework), and the database for storing the data(We will use SQLite Database).
```

Comment:
`The shortener service will be made up of 3 parts; the front-end for interacting with the client(we will create the front-end using React), the backend for creating the server, and providing the API routes(we will create the backend using Django Rest Framework), and the database for storing the data(We will use SQLite Database).`

Make these 3 parts as bulleted points.

Reply:
Both the long URL and the short URL are stored in the database. We are fetching the long URL from the database in the `directUrl` function in the `views.py` file (by fetching the object that has the same short URL as the one provided as a parameter). We are then using this long URL to redirect the client to the correct webpage.